test/Worker/Core.Tests/DurableTaskRegistryTests.Activities.cs

```diff
@@ -0,0 +1,190 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.DurableTask.Worker.Tests;
+
+public partial class DurableTaskRegistryTests
+{
+    [Fact]
+    public void AddActivity_DefaultName_Throws()
+    {
+        DurableTaskRegistry registry = new();
+        Action act = () => registry.AddActivity(default, (IServiceProvider _) => new TestActivity());
```

Comment:
After adding all these overloads (which I think having all of them is great), automatic overload resolution is not great anymore. It is actually resolving _incorrect_ overloads. I would like to improve this to prevent accidentally using the wrong overload. I am thinking the best route is to differentiate some of these by name. At least different names for the lower level _factory_ method and the lambda func-implementation helpers. @cgillum any preference or thoughts on naming?

Reply:
oi...naming is hard. I think the principle to follow is to ensure that users can easily discover the method that they need, so having clarity and consistency is key. For example, if you want to rename the func-implementation helpers, then `AddActivityFunc` and `AddOrchestratorFunc` seem like good choices. They have the same prefixes and are sufficiently descriptive, IMO.

Reply:
Those names sound great to me! Will update with that. Thank you.